src/EA.Weee.DataAccess/DataAccess/EvidenceDataAccess.cs

```diff
@@ -178,9 +179,12 @@ public async Task<int> GetComplianceYearByNotes(List<Guid> evidenceNoteIds)
                                                          n.Status.Value == NoteStatus.Approved.Value &&
                                                          n.ComplianceYear == complianceYear);
 
-            if (searchRef.HasValue)
+            if (!string.IsNullOrWhiteSpace(searchRef))
             {
-                filteredNotes = filteredNotes.Where(n => n.Reference == searchRef.Value);
+                var regex = new Regex("^[E?|e?|][1-9]+");
```

Comment:
I think it matches if searchRef = e3tr4fd3d, is it okay to have letters in the ref?

Reply:
Updated the regex